bCNC/bmain.py

```diff
@@ -0,0 +1,3030 @@
+import sys
+import time
+import traceback
+import webbrowser
+from datetime import datetime
+import tkinter
+from queue import Empty
+from tkinter import (
+    TclError,
+    NO,
+    YES,
+    TRUE,
+    W,
+    E,
+    NW,
+    NE,
+    EW,
+    X,
+    BOTH,
+    LEFT,
+    TOP,
+    RIGHT,
+    BOTTOM,
+    RAISED,
+    SUNKEN,
+    HORIZONTAL,
+    END,
+    NORMAL,
+    DISABLED,
+    Tk,
+    Toplevel,
+    Button,
+    Entry,
+    Frame,
+    Label,
+    Listbox,
+    Text,
+    PhotoImage,
+    Spinbox,
+    LabelFrame,
+    PanedWindow,
+    messagebox,
+)
+
+try:
+    import serial
+except ImportError:
+    serial = None
+    print("testing mode, could not import serial")
+
+import Utils
+
+import bFileDialog
+import CNCCanvas
+
+import rexx
+import Updates
+import tkDialogs
+import tkExtra
+# Load configuration before anything else
+# and if needed replace the  translate function _()
+# before any string is initialized
+from CNC import CNC, WAIT, GCode
+import Ribbon
+import Pendant
+from CNCRibbon import Page
+from ControlPage import ControlPage
+from EditorPage import EditorPage
+from FilePage import FilePage
+from ProbePage import ProbePage
+from Sender import NOT_CONNECTED, STATECOLOR, STATECOLORDEF, Sender
+from TerminalPage import TerminalPage
+from ToolsPage import Tools, ToolsPage
+
+from Helpers import _
+
+Utils.loadConfiguration()
+
+__version__ = Utils.__version__
+__date__ = Utils.__date__
+__author__ = Utils.__author__
+__email__ = Utils.__email__
+
+if not (sys.version_info.major == 3 and sys.version_info.minor >= 8):
+    print("ERROR: Python3.8 or newer is required to run bCNC!!")
+    exit(1)
+
+__platform_fingerprint__ = "".join([
+    f"({sys.platform} ",
+    f"py{sys.version_info.major}.",
+    f"{sys.version_info.minor}.",
+    f"{sys.version_info.micro})"
+])
+
+_openserial = True  # override ini parameters
+_device = None
+_baud = None
+
+MONITOR_AFTER = 200  # ms
+DRAW_AFTER = 300  # ms
+
+RX_BUFFER_SIZE = 128
+
+MAX_HISTORY = 500
+
+FILETYPES = [
+    (
+        _("All accepted"),
+        (
+            "*.ngc",
+            "*.cnc",
+            "*.nc",
+            "*.tap",
+            "*.gcode",
+            "*.dxf",
+            "*.probe",
+            "*.orient",
+            "*.stl",
+            "*.svg",
+        ),
+    ),
+    (_("G-Code"), ("*.ngc", "*.cnc", "*.nc", "*.tap", "*.gcode")),
+    (_("G-Code clean"), ("*.txt")),
+    ("DXF", "*.dxf"),
+    ("SVG", "*.svg"),
+    (_("Probe"), ("*.probe", "*.xyz")),
+    (_("Orient"), "*.orient"),
+    ("STL", "*.stl"),
+    (_("All"), "*"),
+]
+
+geometry = None
+
+# =============================================================================
+# Main Application window
+# =============================================================================
+
+
+class Application(Toplevel, Sender):
+    def __init__(self, master, **kw):
+        Toplevel.__init__(self, master, **kw)
+        Sender.__init__(self)
+
+        if sys.platform == "win32":
```

Comment:
Please remove this code. It is not needed anymore and actually prevents the correct display of the icon